corehq/apps/domain/templates/domain/repeat_record_report.html

```diff
@@ -0,0 +1,124 @@
+/>{% endblock stylesheets %}
+
+{% compress js %}
+{% block js %}{{ block.super }}
+<script src="{% new_static 'codemirror/lib/codemirror.js' %}"></script>
+<script src="{% new_static 'codemirror/mode/xml/xml.js' %}"></script>
+<script src="{% new_static 'codemirror/mode/javascript/javascript.js' %}"></script>
+{% endblock js %}
+{% endcompress %}
+
+{% block js-inline %}{{ block.super }}
+<script>
+    $(document).ready(function() {
+        $('#view-record-payload-modal').on('shown.bs.modal', function(event) {
+            var recordData = $(event.relatedTarget).data(),
+                $modal = $(this);
+
+            $.get("{% url 'repeat_record' domain %}", { record_id: recordData.recordId })
+                .success(function(data) {
+                    var $payload = $modal.find('.payload'),
+                        codeMirror,
+                        contentType = data.content_type,
+                        mode = 'xml';
+
+                    if (contentType === 'text/xml') {
+                        mode = 'xml';
+                    } else if (contentType === 'application/json') {
+                        mode = { name: 'javascript', json: true };
+                    }
+
+                    codeMirror = CodeMirror($payload[0], {
+                        value: data.payload,
+                        mode: mode,
+                        readOnly: true,
+                        lineNumbers: true,
+                        lineWrapping: true
+                    });
+                    codeMirror.refresh();
+
+                })
+                .fail(function(data) {
+                    $modal.find('.modal-body').html('fail')
+                });
+        })
+
+        $('#report-content').on('click', '.resend-record-payload', function() {
+              var $btn = $(this),
+                  recordId = $btn.data().recordId;
+              $btn.disableButton();
+
+              $.post("{% url 'repeat_record' domain %}", { record_id: recordId })
+                  .success(function(data) {
+                      $btn.removeSpinnerFromButton();
+                      if (data.success) {
+                          $btn.text(gettext('Success!'));
+                          $btn.addClass('btn-success');
+                      } else {
+                          $btn.text(gettext('Failed'));
+                          $btn.addClass('btn-danger');
+                          $('#payload-error-modal').modal(
+                              'show')
+                          $(
+                              '#payload-error-modal .error-message'
+                          ).text(data.failure_reason);
+                      }
+
+                  })
+                  .fail(function(data) {
+                      $btn.removeSpinnerFromButton();
+                      $btn.text(gettext('Failed to send'));
+                      $btn.addClass('btn-danger');
+                  });
+          });
+    });
```

Comment:
ouf realize some of the formatting is whack (used an automatic formatter to fix some of the html below). i think for this small case it's ok for this to be inline and not using a library as it's very basic dom manipulation; open to push back though